ci-operator/step-registry/aws/provision/bastionhost/aws-provision-bastionhost-commands.sh

```diff
@@ -0,0 +1,448 @@
+        "contents": {
+          "source": "data:text/plain;base64,"
+        },
+        "mode": 420
+      },
+      {
+        "path": "/opt/registry/auth/htpasswd",
+        "contents": {
+          "source": "data:text/plain;base64,${REGISTRY_PASSWORD_CONTENT}"
+        },
+        "mode": 420
+      },
+      {
+        "path": "/opt/registry/certs/domain.crt",
+        "contents": {
+          "source": "data:text/plain;base64,${REGISTRY_CRT_CONTENT}"
+        },
+        "mode": 420
+      },
+      {
+        "path": "/opt/registry/certs/domain.key",
+        "contents": {
+          "source": "data:text/plain;base64,${REGISTRY_KEY_CONTENT}"
+        },
+        "mode": 420
+      }
+    ],
+    "directories": [
+      {
+        "path": "/srv/squid/log",
+        "mode": 493
+      },
+      {
+        "path": "/srv/squid/cache",
+        "mode": 493
+      },
+      {
+        "path": "/opt/registry/data",
+        "mode": 493
+      }
+    ]
+  },
+  "systemd": {
+    "units": [
+      {
+        "contents": "${PROXY_SERVICE_CONTENT}",
+        "enabled": true,
+        "name": "squid-proxy.service"
+      },
+      {
+        "contents": "${REGISTRY_SERVICE_CONTENT}",
+        "enabled": true,
+        "name": "poc-registry.service"
+      }
+    ]
+  }
+}
+EOF
+
+
+# upload ignition file to s3
+echo "s3://${s3_bucket_name}" > "$SHARED_DIR/bastionhosts3bucket"
+aws --region $REGION s3 mb "s3://${s3_bucket_name}"
+aws --region $REGION s3api put-bucket-acl --bucket "${s3_bucket_name}" --acl public-read
+aws --region $REGION s3 cp ${workdir}/bastion.ign "s3://${s3_bucket_name}/bastion.ign"
+aws --region $REGION s3api put-object-acl --bucket "${s3_bucket_name}" --key "bastion.ign" --acl public-read
+
+
+aws --region $REGION cloudformation create-stack --stack-name ${stack_name} \
+    --template-body file://${workdir}/bastion.yaml \
+    --parameters \
+        ParameterKey=VpcId,ParameterValue="${VpcId}"  \
+        ParameterKey=BastionHostInstanceType,ParameterValue="${BastionHostInstanceType}"  \
+        ParameterKey=Machinename,ParameterValue="${stack_name}"  \
+        ParameterKey=PublicSubnet,ParameterValue="${PublicSubnet}" \
+        ParameterKey=AmiId,ParameterValue="${AmiId}" \
+        ParameterKey=BastionIgnitionLocation,ParameterValue="s3://${s3_bucket_name}/bastion.ign"  &
+
+wait "$!"
+echo "Created stack"
+
+aws --region "${REGION}" cloudformation wait stack-create-complete --stack-name "${stack_name}" &
+wait "$!"
+echo "Waited for stack"
+
+echo "$stack_name" > "${SHARED_DIR}/bastionhoststackname"
+
+INSTANCE_ID="$(aws --region "${REGION}" cloudformation describe-stacks --stack-name "${stack_name}" \
+--query 'Stacks[].Outputs[?OutputKey == `BastionInstanceId`].OutputValue' --output text)"
+echo "Instance ${INSTANCE_ID}"
+
+# to allow log collection during gather:
+# append to proxy bastion host ID to "${SHARED_DIR}/aws-instance-ids.txt"
+echo "${INSTANCE_ID}" >> "${SHARED_DIR}/aws-instance-ids.txt"
```

Comment:
Can't this be written directly to ARTIFACT_DIR?

Reply:
I wrote data into `${ARTIFACT_DIR}` is for testing/debugging only, now I removed those sections, and use a standard way (`${SHARED_DIR}`) to pass values.